airflow/contrib/hooks/sagemaker_hook.py

```diff
@@ -16,299 +16,793 @@
+        :param positions: A list of pairs of (timestamp, skip) which represents the last record
+            read from each stream.
+        :type positions: list
+        :return: A tuple of (stream number, cloudwatch log event).
         """
-        return self.conn.list_hyper_parameter_tuning_job(
-            NameContains=name_contains, StatusEquals=status_equals)
+        positions = positions or {s: Position(timestamp=0, skip=0) for s in streams}
+        event_iters = [self.log_stream(log_group, s, positions[s].timestamp, positions[s].skip)
+                       for s in streams]
+        events = [next(s) if s else None for s in event_iters]
+
+        while some(events):
+            i = argmin(events, lambda x: x['timestamp'] if x else 9999999999)
+            yield (i, events[i])
+            try:
+                events[i] = next(event_iters[i])
+            except StopIteration:
+                events[i] = None
 
-    def create_training_job(self, training_job_config, wait_for_completion=True):
+    def create_training_job(self, config, wait_for_completion=True, print_log=True,
+                            check_interval=30, max_ingestion_time=None):
         """
         Create a training job
-        :param training_job_config: the config for training
-        :type training_job_config: dict
+
+        :param config: the config for training
+        :type config: dict
         :param wait_for_completion: if the program should keep running until job finishes
         :type wait_for_completion: bool
-        :return: A dict that contains ARN of the training job.
+        :param check_interval: the time interval in seconds which the operator
+            will check the status of any SageMaker job
+        :type check_interval: int
+        :param max_ingestion_time: the maximum ingestion time in seconds. Any
+            SageMaker jobs that run longer than this will fail. Setting this to
+            None implies no timeout for any SageMaker job.
+        :type max_ingestion_time: int
+        :return: A response to training job creation
         """
-        if self.use_db_config:
-            if not self.sagemaker_conn_id:
-                raise AirflowException("SageMaker connection id must be present to read \
-                                        SageMaker training jobs configuration.")
-            sagemaker_conn = self.get_connection(self.sagemaker_conn_id)
-
-            config = copy.deepcopy(sagemaker_conn.extra_dejson)
-            training_job_config.update(config)
 
-        self.check_valid_training_input(training_job_config)
+        self.check_training_config(config)
+
+        response = self.get_conn().create_training_job(**config)
+        if print_log:
+            self.check_training_status_with_log(config['TrainingJobName'],
+                                                SageMakerHook.non_terminal_states,
+                                                SageMakerHook.failed_states,
```

Comment:
This is more pythonic: 
```suggestion
                                                self.failed_states,
```

and similarly for all the other uses too please.

Reply:
Though see the comment on `def check_status` first.

Reply:
Updated.